File: tests/java/com/trulioo/sdk/model/NormalizedDatasourceGroupCountryTest.java
package com.trulioo.sdk.model;

import static org.junit.jupiter.api.Assertions.*;

import org.junit.jupiter.api.*;

import java.util.List;
import java.util.Arrays;

public class NormalizedDatasourceGroupCountryTest {
    private NormalizedDatasourceGroupCountry normalizedDatasourceGroupCountry;

    @BeforeEach
    public void beforeEach() {
        normalizedDatasourceGroupCountry = new NormalizedDatasourceGroupCountry();
    }

    @Test
    public void testName() {
        String name = "test";
        normalizedDatasourceGroupCountry.setName(name);

        assertEquals(name, normalizedDatasourceGroupCountry.getName());
        assertEquals(name, new NormalizedDatasourceGroupCountry().name(name).getName());
    }

    @Test
    public void testDescription() {
        String description = "desc";
        normalizedDatasourceGroupCountry.setDescription(description);

        assertEquals(description, normalizedDatasourceGroupCountry.getDescription());
        assertEquals(description, new NormalizedDatasourceGroupCountry().description(description).getDescription());
    }

    @Test
    public void testRequiredFields() {
        normalizedDatasourceGroupCountry.addRequiredFieldsItem(new NormalizedDatasourceField());
        normalizedDatasourceGroupCountry.addRequiredFieldsItem(new NormalizedDatasourceField());
        List<NormalizedDatasourceField> requiredFields = Arrays
                .asList(new NormalizedDatasourceField().fieldName("required"));
        normalizedDatasourceGroupCountry.setRequiredFields(requiredFields);

        assertEquals(requiredFields, normalizedDatasourceGroupCountry.getRequiredFields());
        assertEquals(requiredFields,
                new NormalizedDatasourceGroupCountry().requiredFields(requiredFields).getRequiredFields());
    }

    @Test
    public void testOptionalFields() {
        normalizedDatasourceGroupCountry.addOptionalFieldsItem(new NormalizedDatasourceField());
        normalizedDatasourceGroupCountry.addOptionalFieldsItem(new NormalizedDatasourceField());
        List<NormalizedDatasourceField> optionalFields = Arrays
                .asList(new NormalizedDatasourceField().fieldName("optional"));
        normalizedDatasourceGroupCountry.setOptionalFields(optionalFields);

        assertEquals(optionalFields, normalizedDatasourceGroupCountry.getOptionalFields());
        assertEquals(optionalFields,
                new NormalizedDatasourceGroupCountry().optionalFields(optionalFields).getOptionalFields());
    }

    @Test
    public void testAppendedFields() {
        normalizedDatasourceGroupCountry.addAppendedFieldsItem(new NormalizedDatasourceField());
        normalizedDatasourceGroupCountry.addAppendedFieldsItem(new NormalizedDatasourceField());
        List<NormalizedDatasourceField> appendedFields = Arrays
                .asList(new NormalizedDatasourceField().fieldName("appended"));
        normalizedDatasourceGroupCountry.setAppendedFields(appendedFields);

        assertEquals(appendedFields, normalizedDatasourceGroupCountry.getAppendedFields());
        assertEquals(appendedFields,
                new NormalizedDatasourceGroupCountry().appendedFields(appendedFields).getAppendedFields());
    }

    @Test
    public void testOutputFields() {
        normalizedDatasourceGroupCountry.addOutputFieldsItem(new NormalizedDatasourceField());
        normalizedDatasourceGroupCountry.addOutputFieldsItem(new NormalizedDatasourceField());
        List<NormalizedDatasourceField> outputFields = Arrays
                .asList(new NormalizedDatasourceField().fieldName("output"));
        normalizedDatasourceGroupCountry.setOutputFields(outputFields);

        assertEquals(outputFields, normalizedDatasourceGroupCountry.getOutputFields());
        assertEquals(outputFields, new NormalizedDatasourceGroupCountry().outputFields(outputFields).getOutputFields());
    }

    @Test
    public void testSourceType() {
        String sourceType = "source";
        normalizedDatasourceGroupCountry.setSourceType(sourceType);

        assertEquals(sourceType, normalizedDatasourceGroupCountry.getSourceType());
        assertEquals(sourceType, new NormalizedDatasourceGroupCountry().sourceType(sourceType).getSourceType());
    }

    @Test
    public void testUpdateFrequency() {
        String updateFrequency = "3600";
        normalizedDatasourceGroupCountry.setUpdateFrequency(updateFrequency);

        assertEquals(updateFrequency, normalizedDatasourceGroupCountry.getUpdateFrequency());
        assertEquals(updateFrequency,
                new NormalizedDatasourceGroupCountry().updateFrequency(updateFrequency).getUpdateFrequency());
    }

    @Test
    public void testCoverage() {
        String coverage = "25%";
        normalizedDatasourceGroupCountry.setCoverage(coverage);

        assertEquals(coverage, normalizedDatasourceGroupCountry.getCoverage());
        assertEquals(coverage, new NormalizedDatasourceGroupCountry().coverage(coverage).getCoverage());
    }

    @Test
    public void testEquals() {
        String name = "test";
        NormalizedDatasourceGroupCountry normalizedDatasourceGroupCountry1 = new NormalizedDatasourceGroupCountry()
                .name(name);

        assertEquals(normalizedDatasourceGroupCountry1, normalizedDatasourceGroupCountry1);
        assertEquals(normalizedDatasourceGroupCountry1, new NormalizedDatasourceGroupCountry().name(name));
        assertNotEquals(normalizedDatasourceGroupCountry1, new NormalizedDatasourceGroupCountry().name(name + "1"));
        assertNotEquals(normalizedDatasourceGroupCountry1, null);
    }

    @Test
    public void testHashcode() {
        String name = "test";

        assertEquals(new NormalizedDatasourceGroupCountry().name(name).hashCode(),
                new NormalizedDatasourceGroupCountry().name(name).hashCode());
    }

    @Test
    public void testToString() {
        String name = "test";

        assertEquals(new NormalizedDatasourceGroupCountry().name(name).toString(),
                new NormalizedDatasourceGroupCountry().name(name).toString());
    }
}

File: tests/unit/php/Model/NormalizedDatasourceGroupCountryTest.php
<?php

namespace Trulioo\SDK\Test\Model;

use InvalidArgumentException;
use Trulioo\SDK\Model\NormalizedDatasourceField;
use PHPUnit\Framework\TestCase;
use Trulioo\SDK\Model\NormalizedDatasourceGroupCountry;

class NormalizedDatasourceGroupCountryTest extends TestCase
{
    private NormalizedDatasourceGroupCountry $normalizedDatasourceGroupCountry;

    public function setUp(): void
    {
        $this->normalizedDatasourceGroupCountry = new NormalizedDatasourceGroupCountry([]);
    }

    public function testPropertyName()
    {
        $this->normalizedDatasourceGroupCountry->setName('field');
        $this->assertEquals('field', $this->normalizedDatasourceGroupCountry->getName());

        try {
            $this->normalizedDatasourceGroupCountry->setName(str_repeat('a', 201));
            $this->fail('Expected exception');
        } catch (InvalidArgumentException $e) {
        }
    }

    public function testPropertyDescription()
    {
        $this->normalizedDatasourceGroupCountry->setDescription('test');
        $this->assertEquals('test', $this->normalizedDatasourceGroupCountry->getDescription());

        try {
            $this->normalizedDatasourceGroupCountry->setDescription(str_repeat('a', 501));
            $this->fail('Expected exception');
        } catch (InvalidArgumentException $e) {
        }
    }

    public function testPropertyRequiredFields()
    {
        $this->normalizedDatasourceGroupCountry->setRequiredFields([
            new NormalizedDatasourceField([
                'field_name' => 'field'
            ])
        ]);
        $this->assertEquals(
            [
                new NormalizedDatasourceField([
                    'field_name' => 'field'
                ])
            ],
            $this->normalizedDatasourceGroupCountry->getRequiredFields()
        );
    }

    public function testPropertyOptionalFields()
    {
        $this->normalizedDatasourceGroupCountry->setOptionalFields([
            new NormalizedDatasourceField([
                'field_name' => 'option1'
            ])
        ]);
        $this->assertEquals(
            [
                new NormalizedDatasourceField([
                    'field_name' => 'option1'
                ])
            ],
            $this->normalizedDatasourceGroupCountry->getOptionalFields()
        );
    }

    public function testPropertyAppendedFields()
    {
        $this->normalizedDatasourceGroupCountry->setAppendedFields([
            new NormalizedDatasourceField([
                'field_name' => 'appended1'
            ])
        ]);
        $this->assertEquals(
            [
                new NormalizedDatasourceField([
                    'field_name' => 'appended1'
                ])
            ],
            $this->normalizedDatasourceGroupCountry->getAppendedFields()
        );
    }

    public function testPropertyOutputFields()
    {
        $this->normalizedDatasourceGroupCountry->setOutputFields([
            new NormalizedDatasourceField([
                'field_name' => 'output1'
            ])
        ]);
        $this->assertEquals(
            [
                new NormalizedDatasourceField([
                    'field_name' => 'output1'
                ])
            ],
            $this->normalizedDatasourceGroupCountry->getOutputFields()
        );
    }

    public function testPropertySourceType()
    {
        $this->normalizedDatasourceGroupCountry->setSourceType('source');
        $this->assertEquals('source', $this->normalizedDatasourceGroupCountry->getSourceType());

        try {
            $this->normalizedDatasourceGroupCountry->setSourceType(str_repeat('a', 251));
            $this->fail('Expected exception');
        } catch (InvalidArgumentException $e) {
        }
    }

    public function testPropertyUpdateFrequency()
    {
        $this->normalizedDatasourceGroupCountry->setUpdateFrequency('3600');
        $this->assertEquals('3600', $this->normalizedDatasourceGroupCountry->getUpdateFrequency());

        try {
            $this->normalizedDatasourceGroupCountry->setUpdateFrequency(str_repeat('1', 65));
            $this->fail('Expected exception');
        } catch (InvalidArgumentException $e) {
        }
    }

    public function testPropertyCoverage()
    {
        $this->normalizedDatasourceGroupCountry->setCoverage('25%');
        $this->assertEquals('25%', $this->normalizedDatasourceGroupCountry->getCoverage());

        try {
            $this->normalizedDatasourceGroupCountry->setCoverage(str_repeat('a', 65));
            $this->fail('Expected exception');
        } catch (InvalidArgumentException $e) {
        }
    }
}
